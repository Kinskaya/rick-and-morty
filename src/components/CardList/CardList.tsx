import React from 'react';
import { useSelector } from 'react-redux';

import Card, { ICard } from '../Card/Card';

import '../Card/Card.css';

import { RootState } from '../../redux/store';

/*export interface ICardList {
  info: ICard[];
  loading: boolean;
  error: string;
}*/
const CardList = () => {
  const info = useSelector<RootState>((state) => state.cards.info) as ICard[];
  const error = useSelector<RootState>((state) => state.cards.error) as string;
  const loading = useSelector<RootState>((state) => state.cards.loading) as string;

  if (error) {
    return <h2 className="loading">{error}</h2>;
  }
  if (loading || !info) {
    return <h2 className="loading">Loading...</h2>;
  }
  return (
    <div className="card-container">
      {info.map((item: ICard) => {
        return (
          <Card
            key={item.id}
            id={item.id}
            name={item.name}
            gender={item.gender}
            image={item.image}
            status={item.status}
            species={item.species}
            created={item.created}
            origin={item.origin}
          />
        );
      })}
    </div>
  );
};

export default CardList;
