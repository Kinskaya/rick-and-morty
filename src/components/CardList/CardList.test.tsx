import { render } from '@testing-library/react';
import CardList from './CardList';
import React from 'react';

describe('CardList component', () => {
  it('render correctly', () => {
    const error = 'Sorry, nothing found';
    const { asFragment } = render(<CardList />);
    expect(asFragment()).toMatchSnapshot();
  });
});
