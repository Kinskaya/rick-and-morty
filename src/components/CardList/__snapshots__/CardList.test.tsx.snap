// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardList component render correctly 1`] = `
<DocumentFragment>
  <h2
    class="loading"
  >
    Loading...
  </h2>
</DocumentFragment>
`;
